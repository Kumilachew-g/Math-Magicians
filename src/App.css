.App {
  text-align: center;
}

.welcome-home {
  position: absolute;
  font-size: 22px;
  padding: 0 30px;
  color: #fff;
}

.welcome-home h1 {
  text-align: center;
}

#react-particles-js {
  height: 100vh;
}

.nav-bar {
  display: flex;
  justify-content: space-between;
  padding: 0 20px;
  border-bottom: 1px solid #e0dddd;
  color: #fff;
}

.navbar-links {
  padding: 0 10px;
  font-size: 24px;
  text-decoration: none;
  border-right: 1px solid #f0f0f6;
  color: #fff;
}

.no-match {
  font-size: 32px;
  text-align: center;
}

.quotes {
  position: absolute;
  font-size: 22px;
  padding: 100px 0;
  margin: auto 20%;
  text-align: center;
  color: #fff;
}
