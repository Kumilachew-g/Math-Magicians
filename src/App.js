import React from 'react';
import './App.css';
import Calculator from './components/calculator';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      digitObj: {
        totalOutput: 0,
        nextOutput: 0,
        operation: null,
      },
    };
    this.updateState = this.updateState.bind(this);
  }

  updateState(obj) {
    this.setState({
      digitObj: obj,
    });
  }

  render() {
    const { digitObj } = this.state;
    return (
      <div className="App">
        <Calculator updateState={this.updateState} digitObj={digitObj} />
      </div>
    );
  }
}

export default App;
