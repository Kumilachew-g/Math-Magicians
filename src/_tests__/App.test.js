import App from "../App";
test('expect APP page to match working snapshot', () => {
    const appPage = render(<App />);
    expect(appPage).toMatchSnapshot();
  });