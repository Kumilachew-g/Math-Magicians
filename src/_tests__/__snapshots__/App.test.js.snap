// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect APP page to match working snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav-bar"
      >
        <h1>
          Math Magician
        </h1>
        <ul>
          <a
            class="navbar-links"
            href="/Math-Magician"
          >
            Home
          </a>
          <a
            class="navbar-links"
            href="/Math-Magician/calculator"
          >
            Calculator
          </a>
          <a
            class="navbar-links"
            href="/Math-Magician/quote"
          >
            Qoute
          </a>
        </ul>
      </nav>
      <div
        class="no-match"
      >
        <h3>
          Whoops! Page not found!
        </h3>
        <img
          alt="Not found"
          src="notFound.jpg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav-bar"
    >
      <h1>
        Math Magician
      </h1>
      <ul>
        <a
          class="navbar-links"
          href="/Math-Magician"
        >
          Home
        </a>
        <a
          class="navbar-links"
          href="/Math-Magician/calculator"
        >
          Calculator
        </a>
        <a
          class="navbar-links"
          href="/Math-Magician/quote"
        >
          Qoute
        </a>
      </ul>
    </nav>
    <div
      class="no-match"
    >
      <h3>
        Whoops! Page not found!
      </h3>
      <img
        alt="Not found"
        src="notFound.jpg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
